tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -1089,4 +1348,22 @@ private Path getRootPath() throws Exception {
         return getRootPath(true);
     }
 
```

Comment:
Just a general comment. Have you tested the case where the identity file has been tampered with?

Reply:
Have you tested the case where the signature can't be written to the file for some reason and then verified that the file remains unchanged?

Reply:
> Just a general comment. Have you tested the case where the identity file has been tampered with?

Added 2 tests where identity is tampered.
> Have you tested the case where the signature can't be written to the file for some reason and then verified that the file remains unchanged?

I've tested manually, I'm unable to make a test to get the code into a state where only during writing the signature does it fail. If I try to make that happen, it gets caught in the "verify signature" method instead of the unable to write signature portion, but I am now using the temporary file to make changes before copying that to the final file.